src/modules/deal/dto/deals-request.dto.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsDateString, IsNotEmpty, IsNumber, IsOptional, IsString, MaxLength } from 'class-validator';
+
+export class CreateDealDto {
+  @ApiProperty({
+    example: '0020900035', // Comment on PR if we need a description when it's self evident
+  })
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(10)
+  dealIdentifier: string;
+
+  @ApiProperty({
+    description: 'Deal portfolio is E1',
+    example: 'E1',
+  })
+  @IsString()
+  @IsOptional()
+  @MaxLength(2)
+  portfolioIdentifier = 'E1';
```

Comment:
is `E1` a default value?

Reply:
If yes, then can this please be referred from constants?

Reply:
Please use `@Length(2)` so it validates both `@Min` and `@Max'